system/assets/js/admin/specific/urlInput/urlInput.js

```diff
@@ -2,17 +2,20 @@
 
 	$.fn.presideUrlInput = function() {
 		return this.each( function() {
-			var $input = $( this );
-
-			$input.on( "change", function( e ) {
-				var $this   = $( this )
-				  , $hidden = $this.closest( ".row" ).next( 'input[type="hidden"]' )
-				  , inputId = $hidden.attr( "id" )
+			$( this ).on( "change", function( e ) {
+				var $hidden     = $( this ).closest( ".row" ).next( 'input[type="hidden"]' )
+				  , inputId     = $hidden.attr( "id" )
 				  , $protocol   = $( "#" + inputId + "_protocol" )
 				  , $domainPath = $( "#" + inputId + "_domain_path" )
 				;
 
 				if ( $protocol.val().length > 0 && $domainPath.val().length > 0 ) {
+					var matches = $domainPath.val().match( /^(.*:\/\/)/g );
+
+					$protocol.val( matches ? matches[ 0 ] : "https://" );
```

Comment:
Are we saying that if someone enters a URL _without_ a protocol, we always set the protocol to https whenever a change is detected?

I think the value should only get set if:

1. Protocol is found in the address
2. Or, the protocol is currently empty